### ISSUE TITLE
callback functions cannot be called when training with multiple GPUs

### ISSUE THREAD

**Issue by wangtianlei1998** [ultralytics/ultralytics#16838]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

When training with multiple GPUs, callback functions cannot be called. There are no issues when using a single GPU or CPU

def on_train_epoch_end(trainer):
    print("on_train_epoch_end")
    curr_epoch = trainer.epoch + 1
    text = f"Epoch Number: {curr_epoch}/{trainer.epochs} finished"
    print(text)
model.add_callback("on_train_epoch_end", on_train_epoch_end)
model.train(data="data.yaml", epochs=500, imgsz=1280,freeze=11,device=[0,1])

I trained with this code, but there was no output

### Environment

ultralytics               8.3.8                    pypi_0    pypi
ultralytics-thop          2.0.9                    pypi_0    pypi

### Minimal Reproducible Example

def on_train_epoch_end(trainer):
      print("on_train_epoch_end")
      curr_epoch = trainer.epoch + 1
      text = f"Epoch Number: {curr_epoch}/{trainer.epochs} finished"
      print(text)
model.add_callback("on_train_epoch_end", on_train_epoch_end)

# Train the model
model.train(data="data.yaml", epochs=500, imgsz=1280,freeze=11,device=[0,1])

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @wangtianlei1998, thank you for your interest in Ultralytics 🚀! We recommend a visit to our [Docs](https://docs.ultralytics.com) where you can find comprehensive guides and FAQs.

It looks like you are encountering an issue with callback functions when training with multiple GPUs. This is an automated response to help streamline your experience, and an Ultralytics engineer will assist you shortly.

If this is a 🐛 Bug Report, thank you for including the [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) which will help us debug it.

For more detailed guidance, make sure you are following our [Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) to ensure optimal performance during training.

Join the Ultralytics community where it suits you best. Check out real-time discussions on our [Discord](https://ultralytics.com/discord) 🎧, engage in depth at [Discourse](https://community.ultralytics.com), or explore threads on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Make sure you are using the latest `ultralytics` package. Upgrade in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in verified environments that come with preinstalled dependencies:

- **Notebooks** with GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon**: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing across all major operating systems.

We appreciate your patience and will get back to you with more specific guidance soon!

**Comment by Y-T-G** [comment]
Since multi-GPU launches subprocesses, the custom callbacks don't persist
